django_components/templatetags/component_tags.py

```diff
@@ -457,14 +459,15 @@ def try_parse_as_named_fill_tag_set(
     nodelist: NodeList,
 ) -> Optional[Iterable[NamedFillNode]]:
     result = []
-    seen_name_fexps = set()
+    seen_name_fexps: Set[FilterExpression] = set()
     for node in nodelist:
         if isinstance(node, NamedFillNode):
             if node.name_fexp in seen_name_fexps:
                 raise TemplateSyntaxError(
                     f"Multiple fill tags cannot target the same slot name: "
                     f"Detected duplicate fill tag name '{node.name_fexp}'."
                 )
+            seen_name_fexps.add(node.name_fexp)
```

Comment:
IMPORTANT: I've found that this check with `seen_name_fexps` wasn't working - `seen_name_fexps` was never populated, so the error would never raise.